frankenstein/ingester_lifecycle.go

```diff
@@ -30,23 +32,59 @@ const (
 	infName = "eth0"
 )
 
-// WriteIngesterConfigToConsul writes ingester config to Consul
-func WriteIngesterConfigToConsul(consulClient ConsulClient, listenPort int, numTokens int) error {
-	log.Info("Adding ingester to consul")
+// IngesterRegistration is the registration of an ingester with a register.
+type IngesterRegistration interface {
+	Unregister() error
+}
```

Comment:
We should start following golang's addage or "return structs, accept interfaces":